src/main/java/com/fjb/sunrise/services/impl/UserServiceImpl.java

```diff
@@ -1,8 +1,68 @@
 package com.fjb.sunrise.services.impl;
 
+import com.fjb.sunrise.dtos.requests.RegisterRequest;
+import com.fjb.sunrise.enums.ERole;
+import com.fjb.sunrise.models.User;
+import com.fjb.sunrise.repositories.UserRepository;
 import com.fjb.sunrise.services.UserService;
+import java.time.LocalDateTime;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
 
 @Service
+@RequiredArgsConstructor
 public class UserServiceImpl implements UserService {
+    @Value("${application.admin.default.key}")
+    private String key;
+    private final UserRepository repository;
+    private final PasswordEncoder passwordEncoder;
+
+    @Override
+    public boolean checkRegister(RegisterRequest registerRequest) {
+        if (registerRequest.getEmail().isEmpty()
```

Comment:
this validations maybe check in the controller, like registerRequest, we use spring validation to check, the service is only for business logic